zanata-war/src/main/java/org/zanata/events/DocStatsEvent.java

```diff
@@ -0,0 +1,58 @@
+package org.zanata.events;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.zanata.common.ContentState;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import javax.annotation.Nullable;
+
+/**
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@AllArgsConstructor
+@Getter
+@EqualsAndHashCode
+public class DocStatsEvent {
+    private final DocumentLocaleKey key;
+
+    private final Long projectVersionId;
+
+    /**
+     * Updated content states with word counts
+     */
+    private final Map<ContentState, Integer> contentStates;
+
+    private final Long lastModifiedTargetId;
+
+    public static Map<ContentState, Integer> updateContentState(
+        @Nullable Map<ContentState, Integer> contentStates, ContentState newState,
+        ContentState previousState, Long longWordCount) {
```

Comment:
Alex, HTextFlow.getWordCount() is marked NotNull. I think the value can
only be null during initialisation. Once the HTextFlow has been
initialised, wordCount will not be null.

I would switch it all to Long/long rather than Integer if possible.


On 5 May 2016 at 07:47, Alex Eng <notifications@github.com> wrote:

> In zanata-war/src/main/java/org/zanata/events/DocStatsEvent.java
> <https://github.com/zanata/zanata-server/pull/1137#discussion_r62120533>:
>
> > +@EqualsAndHashCode
> > +public class DocStatsEvent {
> > +    private final DocumentLocaleKey key;
> > +
> > +    private final Long projectVersionId;
> > +
> > +    /**
> > +     * Updated content states with word counts
> > +     */
> > +    private final Map<ContentState, Integer> contentStates;
> > +
> > +    private final Long lastModifiedTargetId;
> > +
> > +    public static Map<ContentState, Integer> updateContentState(
> > +        @Nullable Map<ContentState, Integer> contentStates, ContentState newState,
> > +        ContentState previousState, Long longWordCount) {
>
> @seanf <https://github.com/seanf> it's a nullable Long from HTextFlow but
> I don't want to have Null wordcount in the webhook event. Perhaps need to
> accept Integer in the method. What do you think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1137/files/fdee8c13c56ff20a1ffa399056bb04cb6e19832c#r62120533>
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat